packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx

```diff
@@ -26,6 +29,160 @@ const StyledEditable = styled(Editable)`
   }
 `;
 
+const DragItem = styled(Flex)`
+  // block styles
+  & > *:nth-child(2) {
+    width: 100%;
+    opacity: inherit;
+  }
+  &:hover {
+    & > div {
+      visibility: visible;
+      opacity: inherit;
+    }
+  }
+  &:active {
+    opacity: 0.5;
+  }
+`;
+
+const DragButton = styled(Flex)`
+  cursor: pointer;
+  visibility: hidden;
+  &:hover {
+    background: ${({ theme }) => theme.colors.neutral200};
+  }
+  &:active {
+    cursor: grabbing;
+    background: ${({ theme }) => theme.colors.neutral200};
+  }
+`;
+
+const NOT_DRAGGABLE_ITEMS = ['list'];
+const NOT_DROPPABLE_ITEMS = ['list'];
+
+type DragAndDropElementProps = {
+  children: RenderElementProps['children'];
+  index: Array<number>;
+  disabled: boolean;
+  canDrag: boolean;
+  canDrop: boolean;
+  name: string;
+  setLiveText: (text: string) => void;
+};
+
+const DragAndDropElement = ({
+  children,
+  index,
+  disabled,
+  canDrag,
+  canDrop,
+  name,
+  setLiveText,
+}: DragAndDropElementProps) => {
+  const { editor } = useBlocksEditorContext('drag-and-drop');
+  const { formatMessage } = useIntl();
+
+  const handleMoveBlock = React.useCallback(
+    (newIndex: Array<number>, currentIndex: Array<number>) => {
+      const [newNode] = Editor.node(editor, newIndex);
+      const [draggedNode] = Editor.node(editor, currentIndex);
+
+      Transforms.moveNodes(editor, {
+        at: currentIndex,
+        to: newIndex,
+      });
+
+      // Add 1 to the index for the live text message
```

Comment:
Are you sure that's needed? I feel like that's adding complexity for not much. Also when saving the content if you have validation errors, the error message is the actual Slate path, so adding +1 here would not be consistent

Reply:
it's just for the screen reader message as index starts with 0.